test/doc/traits/struct_support.cpp

```diff
@@ -0,0 +1,67 @@
+#include <eve/product_type.hpp>
+#include <eve/wide.hpp>
+
+// position is a product type made of 3 floats
+struct position : eve::struct_support<position, float, float, float>
+{
+  // To simplify member access, we can wrap access to the product type
+  // elements via a named friend accessor.
+  friend decltype(auto) x(eve::like<position> auto&& self)
+  {
+    return get<0>(std::forward<decltype(self)>(self));
+  }
+
+  friend decltype(auto) y(eve::like<position> auto&& self)
+  {
+    return get<1>(std::forward<decltype(self)>(self));
+  }
+
+  friend decltype(auto) z(eve::like<position> auto&& self)
+  {
+    return get<2>(std::forward<decltype(self)>(self));
+  }
+
+  // Defines += over position to automatically generates +
+  friend auto& operator+= ( eve::like<position> auto& self
+                          , eve::like<position> auto const& other
+                          )
+  {
+    x(self) += x(other);
+    y(self) += y(other);
+    z(self) += z(other);
+    return self;
+  }
+
+  // Defines -= over position to automatically generates -
+  friend auto& operator-= ( eve::like<position> auto& self
+                          , eve::like<position> auto const& other
+                          )
+  {
+    x(self) -= x(other);
+    y(self) -= y(other);
+    z(self) -= z(other);
+    return self;
+  }
```

Comment:
Bad example I think. Position + position is not a thing. Position - position is a vector. 

How about vector?

Reply:
Done